src/hotspot/share/gc/shenandoah/shenandoahHeap.inline.hpp

```diff
@@ -261,6 +261,19 @@ inline oop ShenandoahHeap::evacuate_object(oop p, Thread* thread) {
   assert(ShenandoahThreadLocalData::is_evac_allowed(thread), "must be enclosed in oom-evac scope");
 
   size_t size = p->size();
+  ShenandoahGeneration target_gen = heap_region_containing(p)->generation();
+  if (target_gen == YOUNG_GEN) {
+    markWord mark = p->mark();
+    if (mark.is_marked()) {
```

Comment:
This could probably use ShenandoahForwarding::is_marked() instead.

Reply:
Fixed.